File: octave/roots/bisect_test.m
% TEST: Bisection method

format long

display('---------------------------------------------------------------------')
f = inline('sin(x)-(cos(x))^2','x')
[x,steps] = bisect(-1,1,f,eps)
err = abs(feval(f,x))

display('---------------------------------------------------------------------')
f = inline('x^2-8*x+2','x')
[x,steps] = bisect(0,5,f,eps)
err = abs(feval(f,x))

display('---------------------------------------------------------------------')
f = inline('2^x-(cos(x+3))^2+exp(x+2)','x')
[x,steps] = bisect(-10,10,f,eps)
err = abs(feval(f,x))

File: python/bisect_test.py
# Bisection method root finding tests

from bisect import bisect
from math import sin,cos,exp
import numpy as np

# Get machine precision
eps = np.finfo(float).eps

################################################################################

print('-----------------------------------------------------------------------')
print('f(x) = sin(x)-cos(x)^2')
f = lambda x: sin(x)-cos(x)**2
x,steps = bisect(-1,1,f,eps)
print(" x = {}\n steps = {}\n error = {}".format(x,steps,f(x)))

################################################################################

print('-----------------------------------------------------------------------')
print('f(x) = x^2-8*x+2')
f = lambda x: x**2-8*x+2
x,steps = bisect(0,5,f,eps)
print(" x = {}\n steps = {}\n error = {}".format(x,steps,f(x)))

################################################################################

print('-----------------------------------------------------------------------')
print('f(x) = 2**x-cos(x+3)**2+exp(x+2)')
f = lambda x: 2**x-(cos(x+3))**2+exp(x+2)
x,steps = bisect(-10,10,f,eps)
print(" x = {}\n steps = {}\n error = {}".format(x,steps,f(x)))
